### PR TITLE
Fix failures while executing `make test-e2e`

### PR DESCRIPTION
**Pull request by sairameshv**
Observed that the `make test-e2e` command fails due to unavailability of the cert-manager-webhook service with the following error log snippet.
```
[FAILED] Unexpected error:
      <*errors.errorString | 0xc00012c550>: 
      make deploy IMG=quay.io/amalvank/instaslicev2-controller:latest failed with error: (exit status 2) make[1]: Entering directory '/home/svanka/go/src/instaslice-operator'
      test -s /home/svanka/go/src/instaslice-operator/bin/controller-gen && /home/svanka/go/src/instaslice-operator/bin/controller-gen --version | grep -q v0.14.0 || \
      GOBIN=/home/svanka/go/src/instaslice-operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
      /home/svanka/go/src/instaslice-operator/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
      cd config/manager && /home/svanka/go/src/instaslice-operator/bin/kustomize edit set image controller=quay.io/amalvank/instaslicev2-controller:latest daemonset=quay.io/amalvank/instaslicev2-daemonset:latest
      /home/svanka/go/src/instaslice-operator/bin/kustomize build config/default | kubectl apply -f -
      Warning: resource namespaces/instaslice-operator-system is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
      namespace/instaslice-operator-system configured
      customresourcedefinition.apiextensions.k8s.io/instaslices.inference.codeflare.dev configured
      serviceaccount/instaslice-operator-controller-manager created
      role.rbac.authorization.k8s.io/instaslice-operator-leader-election-role created
      clusterrole.rbac.authorization.k8s.io/instaslice-operator-manager-role created
      clusterrole.rbac.authorization.k8s.io/instaslice-operator-metrics-reader created
      clusterrole.rbac.authorization.k8s.io/instaslice-operator-proxy-role created
      rolebinding.rbac.authorization.k8s.io/instaslice-operator-leader-election-rolebinding created
      clusterrolebinding.rbac.authorization.k8s.io/instaslice-operator-manager-rolebinding created
      clusterrolebinding.rbac.authorization.k8s.io/instaslice-operator-proxy-rolebinding created
      service/instaslice-operator-controller-manager-metrics-service created
      service/instaslice-operator-webhook-service created
      deployment.apps/instaslice-operator-controller-manager created
      daemonset.apps/instaslice-operator-controller-daemonset created
      mutatingwebhookconfiguration.admissionregistration.k8s.io/instaslice-operator-mutating-webhook-configuration created
      Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "webhook.cert-manager.io": failed to call webhook: Post "https://cert-manager-webhook.cert-manager.svc:443/validate?timeout=30s": dial tcp 10.96.164.237:443: connect: connection refused
      Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "webhook.cert-manager.io": failed to call webhook: Post "https://cert-manager-webhook.cert-manager.svc:443/validate?timeout=30s": dial tcp 10.96.164.237:443: connect: connection refused
      make[1]: *** [Makefile:230: deploy] Error 1
      make[1]: Leaving directory '/home/svanka/go/src/instaslice-operator'
```

This change waits until the cert-manager-webhook pod is in `Running` state before performing the e2e test.

/cc @harche @asm582 @mamy-CS

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -57,6 +57,36 @@ var _ = Describe("controller", Ordered, func() {
 		outputCm, err := cmdCm.CombinedOutput()
 		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to deploy cert manager: %s", outputCm))
 
+		By("validating the cert-manager-webhook pod is running as expected")
```
Comment (asm582) [inline test/e2e/e2e_test.go:60]: Thanks, Test `should run successfully` has a similar code block, should we make similar pieces into a function and reference it in both the tests?

Comment (mamy-CS) [reply test/e2e/e2e_test.go:60]: Yes, that's a good idea. There is redundancy there.

Comment (sairameshv) [reply test/e2e/e2e_test.go]: Done

**Review by asm582 — APPROVED** [review]

**Pull request merged**